uast/transformer/ops.go

```diff
@@ -746,11 +746,23 @@ func (op opScope) Construct(st *State, n nodes.Node) (nodes.Node, error) {
 // Field is an operation on a specific field of an object.
 type Field struct {
 	Name string // name of the field
-	// Optional can be set to make a field optional. Provided string is used as a variable name to the state of the field.
-	// Note that "optional" means that the field may not exists in the object, and it does not mean that the field can be nil.
+	// Optional can be set to make a field optional. Provided string is used as a variable
+	// name to the state of the field. Note that "optional" means that the field may not
+	// exists in the object, and it does not mean that the field can be nil.
 	// To handle nil fields, see Opt operation.
 	Optional string
-	Op       Op // operation used to check/construct the field value
+	// Drop the field if it exists. If Op is set, it will be called to check the value
+	// before dropping it. If the check fails, the whole transform will be canceled.
+	// This is the preferred way for dropping fields. Avoid using Drop without Op.
```

Comment:
I don't really understand the "Avoid using Drop without Op" part. Does it means that `Op: Any()` should be used anyway? If that's the case: could it be implicit?

Reply:
It sets `Op: Any()` by default if it's not specified and `Drop` is set.

The comment is there because we should avoid `Any`... at any cost :) So it's better to set `Drop: true` and `Op: Bool(true)` for example, so you explicitly specify when the field can be dropped.

Reply:
I think using Drop is specific enough. The comment could add "Please note that you should avoid dropping fields in general unless there is not reasonable alternative".

Reply:
Now that I understand how the Op could be used, I would rephrase as:

```
// Drop the field if it exists (so Optional is implied). If Op is set, it will be called to check the value
// and only drop the field if the value matches it. If it's not set, the field will always be dropped
// but please note that you should avoid dropping fields in this way if there is no
// reasonable alternative.
```